src/NServiceBus.Heartbeat/HeartbeatSender.cs

```diff
@@ -30,43 +30,46 @@ protected override Task OnStart(IMessageSession session, CancellationToken cance
             stopSendingHeartbeatsTokenSource = new CancellationTokenSource();
 
             // don't block here since StartupTasks are executed synchronously.
-            _ = SendEndpointStartupMessage(DateTime.UtcNow, stopSendingHeartbeatsTokenSource.Token);
+            _ = SendEndpointStartupMessageAndSwallowExceptions(DateTime.UtcNow, stopSendingHeartbeatsTokenSource.Token);
 
             Logger.Debug($"Start sending heartbeats every {heartbeatInterval}");
 
             _ = Task.Run(async () =>
                 {
-                    while (!stopSendingHeartbeatsTokenSource.IsCancellationRequested)
+                    try
                     {
-                        try
+                        while (true)
                         {
-                            await Task.Delay(heartbeatInterval, stopSendingHeartbeatsTokenSource.Token).ConfigureAwait(false);
-
-                            var message = new EndpointHeartbeat
+                            try
                             {
-                                ExecutedAt = DateTime.UtcNow,
-                                EndpointName = endpointName,
-                                Host = hostInfo.DisplayName,
-                                HostId = hostInfo.HostId
-                            };
+                                await Task.Delay(heartbeatInterval, stopSendingHeartbeatsTokenSource.Token).ConfigureAwait(false);
 
-                            await backend.Send(message, ttlTimeSpan, dispatcher, stopSendingHeartbeatsTokenSource.Token).ConfigureAwait(false);
-                        }
-                        catch (OperationCanceledException ex)
-                        {
-                            if (stopSendingHeartbeatsTokenSource.IsCancellationRequested)
-                            {
-                                Logger.Debug("Heartbeat sending cancelled.", ex);
+                                var message = new EndpointHeartbeat
+                                {
+                                    ExecutedAt = DateTime.UtcNow,
+                                    EndpointName = endpointName,
+                                    Host = hostInfo.DisplayName,
+                                    HostId = hostInfo.HostId
+                                };
+
+                                await backend.Send(message, ttlTimeSpan, dispatcher, stopSendingHeartbeatsTokenSource.Token).ConfigureAwait(false);
                             }
-                            else
+#pragma warning disable PS0019 // Do not catch Exception without considering OperationCanceledException - considered in block
```

Comment:
I think this may be common enough that the diagnostic should not be reported if the `catch` block mentions `OperationCanceledException`.